### ISSUE TITLE
more solid docker tag name for maven project

### ISSUE THREAD

**Issue by aludwiko** [lightbend/kalix-jvm-sdk#1946]
Currently, the docker tag name contains a timestamp.
```
    <kalixContainerRegistry>kcr.us-east-1.kalix.io</kalixContainerRegistry>
    <kalixOrganization>acme</kalixOrganization>
    <dockerImage>${kalixContainerRegistry}/${kalixOrganization}/${project.artifactId}</dockerImage>
    <dockerTag>${project.version}-${build.timestamp}</dockerTag>
```
It's very problematic when building a deployment pipeline with deployment as a separate step (an example from github actions):
```
      - name: Publish docker
        run: |
          mvn deploy --no-transfer-progress
      - name: Deploy
        run: |
          export dockerImage=...
          export dockerTag=`mvn help:evaluate -Dexpression=dockerTag`
          kalix service deploy $SERVICE_NAME $dockerImage:$dockerTag
```
because calling:
```
mvn help:evaluate -Dexpression=dockerTag
```
will return a different output each time. 

My solution was to use a git commit id from the `git-commit-id-maven-plugin` plugin (similar approach to sbt). The full example is here: https://github.com/aludwiko/cfp-notifier/pull/2/files#diff-d820e80d7fb165181c2f87c9ae115f4050ce9e52d559d4acf418ccab322b2ec9

If that's a good direction, I could update all the mvn-based samples.

**Comment by johanandren** [comment]
The reason that we use timestamp across the samples is that k8 will not always pick up latest from re-used tag and we saw users having problems with deploying new versions of their projects. A unique timestamp-tag makes sure that the version of the image deployed is the intended one.

**Comment by johanandren** [comment]
Does commit id work if you iterate on a project without comitting, but redeploying several times (does it add a +n or something?)

**Comment by aludwiko** [comment]
Unfortunately not. Here is everything that could be part of the description: https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/master/docs/git-describe.md

In that case, maybe we could give some hints to the users in docs/pom.xml samples with a link to the plugin.

**Comment by johanandren** [comment]
Or in the CI/CD documentation perhaps? I guess another option could be to explicitly specify a tag from somewhere else, the CI run or the git tag for example when triggering deploy from a "release" tag like we do for many projects.

**Comment by aludwiko** [comment]
Yeah, we could clarify that users can apply different versioning/tagging strategies and slightly warn the timestamp might not be a good idea for production setup.
